src/homework4/task5.py

```diff
@@ -0,0 +1,64 @@
+# Task 5 — Каждый из N школьников некоторой школы знает Mi языков.
+# Определите, какие языки знают все школьники и языки, которые знает хотя бы
+# один из школьников.
+# Входные данные
+# Первая строка входных данных содержит количество школьников N. Далее идет N
+# чисел Mi, после каждого из чисел идет Mi строк, содержащих названия языков,
+# которые знает i-й школьник.
+# Пример входных данных:
+# 3          # N количество школьников
+# 2          # M1 количество языков первого школьника
+# Russian    # языки первого школьника
+# English
+# 3          # M2 количество языков второго школьника
+# Russian
+# Belarusian
+# English
+# 3
+# Russian
+# Italian
+# French
+#
+#
+# Выходные данные
+# В первой строке выведите количество языков, которые знают все школьники.
+# Начиная со второй строки - список таких языков. Затем - количество языков,
+# которые знает хотя бы один школьник, на следующих строках - список таких
+# языков.
+
+import copy
+
+languages = []
+data_about_students = {}
+
+for students in range(int(input("Number of students: "))):
+    for num_of_languages in range(int(input("Languages of a new student: "))):
+        language = input("Language learned: ")
+        languages.append(language)
+        copy_for_dict = copy.deepcopy(languages)
```

Comment:
why do we need deepcopy here if you can just create new one at the beginning of each iteration?